src/tags/control/Taxonomy/Taxonomy.js

```diff
@@ -317,21 +420,36 @@ const HtxTaxonomy = observer(({ item }) => {
   };
 
   return (
-    <div className="taxonomy" style={{ ...style, ...visibleStyle }}>
-      {(item.loading && isFF(FF_DEV_3617)) ? (
-        <div className="lsf-taxonomy">
-          <Spin size="small"/>
-        </div>
-      ) : (
-        <Taxonomy
+    <div style={{ display: 'grid', gridTemplate: 'auto/1fr 1fr' }}>
+      {isFF(FF_TAXONOMY_ASYNC) ? (
+        <NewTaxonomy
           items={item.items}
           selected={item.selected}
           onChange={item.onChange}
+          onLoadData={item.loadItems}
           onAddLabel={item.userLabels && item.onAddLabel}
           onDeleteLabel={item.userLabels && item.onDeleteLabel}
           options={options}
           isEditable={!item.isReadOnly()}
         />
+      ) : (
+        <div className="taxonomy" style={{ ...style, ...visibleStyle }}>
+          {(item.loading && isFF(FF_DEV_3617)) ? (
+            <div className="lsf-taxonomy">
+              <Spin size="small"/>
+            </div>
+          ) : (
+            <Taxonomy
+              items={item.items}
+              selected={item.selected}
+              onChange={item.onChange}
+              onAddLabel={item.userLabels && item.onAddLabel}
+              onDeleteLabel={item.userLabels && item.onDeleteLabel}
+              options={options}
+              isEditable={!item.isReadOnly()}
+            />
+          )}
+        </div>
       )}
     </div>
```

Comment:
lets use cem rather than native html elements

Reply:
BEM lib doesn't support global classnames for now, so I just improved tags hierarchy as much as I could. thanks!